Linux support

It works (and I even finished the first level, yay) if I delete part of `supaplex.c` in the `if (supportsVirtualKeyboard())` branch. Without that deletion, the error message is: `supaplex.c:(.text+0xa09e): undefined reference to 'inputVirtualKeyboardText'`

--- a/.github/workflows/build-nightly.yml
+++ b/.github/workflows/build-nightly.yml
@@ -4,6 +4,18 @@ on:
     tags:
       - nightly
 jobs:
+  build-linux:
+    runs-on: ubuntu-18.04
+    steps:
+    - uses: actions/checkout@v1
+    - name: Install dependencies
+      run: |
+        ./linux/ci-prepare.sh
+      shell: bash
+    - name: Build OpenSupaplex
+      run: |
+        ./linux/ci-build.sh
+      shell: bash
   build-macos:
     runs-on: macos-latest
     steps:

--- a/.github/workflows/build-release.yml
+++ b/.github/workflows/build-release.yml
@@ -4,6 +4,15 @@ on:
     tags:
       - v*
 jobs:
+  build-linux:
+    runs-on: ubuntu-18.04
+    steps:
+    - uses: actions/checkout@v1
+    - name: Build OpenSupaplex for Ubuntu
+      run: |
+        ./linux/ci-prepare.sh
+        ./linux/ci-build.sh
+      shell: bash
   build-macos:
     runs-on: macos-latest
     steps:

--- a/README.md
+++ b/README.md
@@ -20,6 +20,7 @@ the instructions in [cilliemalan's repo](https://github.com/cilliemalan/supaplex
 Other than that, **OpenSupaplex** can be built for any of the following platforms:
 - macOS
 - Windows
+- Linux
 - PSP
 - PS Vita
 - PlayStation 3

--- a/linux/Makefile
+++ b/linux/Makefile
@@ -1,4 +1,4 @@
-src = $(wildcard ../src/*.c) $(wildcard ../src/sdl2/*.c) $(wildcard ../src/sdl_common/*.c) ../src/lib/ini/ini.c
+src = $(wildcard ../src/*.c) $(wildcard ../src/sdl2/*.c) $(wildcard ../src/sdl_common/*.c) ../src/lib/ini/ini.c ../src/null/virtualKeyboard.c
 obj = $(src:.c=.o)
 
 LDFLAGS = -lSDL2_mixer -lvorbis -logg `sdl2-config --libs` -lm

--- a/linux/ci-prepare.sh
+++ b/linux/ci-prepare.sh
@@ -1,7 +1,3 @@
 #!/bin/bash
 
-# Linux
-# sudo apt-get update && sudo apt-get install libsdl2-dev libsdl2-mixer-dev libvorbis-dev
-
-# macOS
-brew install sdl2 sdl2_mixer libvorbis libogg
+sudo apt-get update && sudo apt-get install libsdl2-dev libsdl2-mixer-dev libvorbis-dev

